workspaces/arborist/lib/override-set.js

```diff
@@ -44,6 +44,43 @@ class OverrideSet {
     }
   }
 
+  childrenAreEqual (other) {
+    if (this.children.size !== other.children.size) {
+      return false
+    }
+    for (const [key] of this.children) {
+      if (!other.children.has(key)) {
+        return false
+      }
+      if (this.children.get(key).value !== other.children.get(key).value) {
+        return false
+      }
+      if (!this.children.get(key).childrenAreEqual(other.children.get(key))) {
+        return false
+      }
+    }
+    return true
+  }
```

Comment:
I think it's more readable this way.

Reply:
Yeah, it may also be more extendable too. Just an observation that it can be hoisted out into a helper function and not necessarily live on the class directly